Userland/Libraries/LibC/unistd.cpp

```diff
@@ -975,8 +975,49 @@ int unveil(char const* path, char const* permissions)
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/getpass.html
 char* getpass(char const* prompt)
 {
-    dbgln("FIXME: getpass('{}')", prompt);
-    TODO();
+    int tty = open("/dev/tty", O_RDWR | O_NOCTTY | O_CLOEXEC);
+    if (tty < 0)
+        return nullptr;
+
+    struct termios backup{};
+    if (tcgetattr(tty, &backup) < 0)
+        return nullptr;
+
+    static char password[PASS_MAX];
```

Comment:
One of the earlier comments was suggesting to `memset(password, 0, sizeof(password)` before reading into it. Since it's static data, there might be junk leftover from a previous call in the buffer.

Reply:
I think this function should not concern itself with that, because we overwrite that anyway with a new null-terminated string. If the caller wants the the password erased, they should do so.

Linux man-pages says: "The calling process should zero the password as soon as possible to avoid leaving the cleartext password visible in the process's address space."